wave_lang/kernel/wave/templates/gemm.py

```diff
@@ -339,3 +339,372 @@ def k_loop(acc: tkl.Register[M, N, tkl.f32]) -> tkl.Register[M, N, tkl.f32]:
     }
 
     return persistent_gemm, hyperparams
+
+
+def get_streamk_gemm_kernel(
+    shape: tuple[int, int, int],
+    mfma_variant: MMAType,
+    threads_per_wave: int = 64,
+    block_shape: Optional[tuple[int, int, int]] = None,
+    num_ctas: Optional[int] = None,
+):
+    """
+    Creates a Stream-K GEMM kernel that distributes work more evenly across CTAs.
+    Partitions work at the granularity of K-loop iterations
+    Spinlock approach
+    """
+    import sympy
+    from wave_lang.kernel._support.indexing import sym
+    from wave_lang.kernel._support.dtype import f16, f32, i32
+    from wave_lang.kernel.lang.wave_types import Memory, Register
+
+    if not block_shape:
+        block_shape = (128, 256, 64)
+
+    m, n, k = shape
+    # note: block_m must equal 128, since it matches with number threads along the cta dim i.e. (128, 2, 1)
```

Comment:
Why does block_m need to be 128? Why can it not be higher or lower? Why can 1 thread not be responsible for more rows or less?